src/ops/utils/ExportImportUtils.ts

```
@@ -118,7 +118,8 @@ export function saveToFile(type, data, identifier, filename) {
   }
   fs.writeFile(filename, JSON.stringify(exportData, null, 2), (err) => {
     if (err) {
-      return printMessage(`ERROR - can't save ${type} to file`, 'error');
+      printMessage(`ERROR - can't save ${type} to file`, 'error');
+      throw err;
```

throwing from within a callback here is silently hidden but the fs module. What is the correct way to propagate errors (and test them) here?

---

everyone seems to recommend using the promise based fs module that allows throwing these exceptions... but do we really want to make this change everywhere?

---

Let's consider refactoring this function to be `async` and see what the blast radius would be.

---

It's used a lot of places but also mostly in async places. Let's see.

---

We actually no longer need to change this as we are not calling the export/backup anymore.